src/workerd/api/actor-state.c++

```diff
@@ -735,6 +735,10 @@ kj::Promise<kj::String> DurableObjectStorage::onNextSessionRestoreBookmark(kj::S
   return cache->onNextSessionRestoreBookmark(bookmark);
 }
 
+kj::Promise<void> DurableObjectStorage::waitForBookmark(kj::String bookmark) {
+  return cache->waitForBookmark(bookmark);
```

Comment:
Can this safely be called multiple times from javascript? As it is here, every call on this will result in a new JavaScript promise backed by a kj::Promise... specifically, if I do..

```js
const p1 = whatever.waitForBookmark('...');
const p2 = whatever.waitForBookmark('...');
```

Then `p1 !== p2`

Should this always return the same JS promise when the bookmark id is the same and is already being waited on?

Reply:
The internal PR waits on the same forked promise, so in production this will wait on the same thing.  `p1` will still be `!== p2`, is that a problem?

Reply:
Not a problem just something to be aware of. Creating multiple promises that wait on the same thing can add performance cost.

Reply:
I think this is fine as-is. Don't think people are likely to call it multiple times in a row like that.